src/metadata.py

```diff
@@ -0,0 +1,252 @@
+            dt_utc = dt_local.astimezone(timezone.utc)
+            exif_dict["GPS"][piexif.GPSIFD.GPSDateStamp] = dt_utc.strftime("%Y:%m:%d")
+            # GPSTimeStamp is array of rationals: [hour, minute, second]
+            h, m, s = dt_utc.hour, dt_utc.minute, dt_utc.second
+            exif_dict["GPS"][piexif.GPSIFD.GPSTimeStamp] = [
+                (h, 1), (m, 1), (s, 1)
+            ]
+
+        # Dump and insert
+        exif_bytes = piexif.dump(exif_dict)
+        piexif.insert(exif_bytes, str(image_path))
+
+
+    except Exception as e:
+        print(f"Failed to set EXIF data for {image_path.name}: {e}")
+
+
+def set_video_metadata(video_path: Path, memory: Memory):
+    """
+    Set video metadata using ffmpeg without re-encoding.
+
+    - Timestamps: Written in UTC using ISO 8601 `YYYY-MM-DDTHH:MM:SSZ` for
+        `creation_time` and a generic `date` tag to maximize cross-platform compatibility.
+    - Software/Make: Identify "Snapchat" as source using common tags (©too/software/Make).
+    - Location: When available, writes Apple Photos-compatible `location` and `location-eng`
+        in ISO 6709 format (±lat±lon+alt/).
+
+    Filesystem timestamps (mtime/atime) are NOT set here; they are applied centrally
+    in `_apply_metadata_to_path(...)` and are forced to UTC.
+    """
+    try:
+        # Prepare creation time in UTC (force timezone-aware -> UTC)
+        # Use ISO 8601 with explicit UTC designator 'Z'
+        dt_utc = memory.date.astimezone(timezone.utc)
+        iso_time = dt_utc.strftime("%Y-%m-%dT%H:%M:%S") + "Z"
+
+        # Base metadata arguments with application source using ©too tag (QuickTime software tag)
+        metadata_args = [
+            "-metadata", f"creation_time={iso_time}",
+            "-metadata", f"date={iso_time}",  # Generic date in UTC for cross-platform compatibility
+            "-metadata", "©too=Snapchat",  # QuickTime software tag - standardized for app identification
+            "-metadata", "software=Snapchat",  # Generic software tag for non-Apple players
+            "-metadata", "Make=Snapchat",  # Camera device/source
+        ]
+
+        # Do not set comment via ffmpeg; XMP dc:description applied later via exiftool if available
+
+        # Add location if available
+        if memory.latitude is not None and memory.longitude is not None:
+            lat = f"{memory.latitude:+.4f}"
+            lon = f"{memory.longitude:+.4f}"
+            alt = getattr(memory, "altitude", 0.0)
+            iso6709 = f"{lat}{lon}+{alt:.3f}/"
+
+            # Apple Photos-compatible fields
+            metadata_args += [
+                "-metadata", f"location={iso6709}",
+                "-metadata", f"location-eng={iso6709}",
+            ]
+
+        # Temporary output file
+        temp_path = video_path.with_suffix(".temp.mp4")
+
+        # Run ffmpeg: copy streams, inject metadata
+        subprocess.run(
+            [
+                "ffmpeg",
```

Comment:
Was just trying out this PR and had a problem when passing my ffmpeg path, I believe this should be read from the config rather than hardcoded?

```suggestion
                config.ffmpeg_path,
```